module/modmanager/manager.go

```diff
@@ -595,23 +594,31 @@ func (mgr *Manager) ValidateConfig(ctx context.Context, conf resource.Config) ([
 	return resp.Dependencies, nil
 }
 
-// ResolveImplicitDependenciesInConfig modifies the config diff to add implicit dependencies to changed resources
-// and updates modular resources whose module has been changed with new implicit deps and adds them to conf.Modified.
+// ResolveImplicitDependenciesInConfig mutates the passed in diff to add modular implicit dependencies to added
+// and modified resources. It also puts modular resources whose module has been modified in conf.Added if they
+// are not already there.
 func (mgr *Manager) ResolveImplicitDependenciesInConfig(ctx context.Context, conf *config.Diff) error {
 	mgr.mu.RLock()
 	defer mgr.mu.RUnlock()
-	// Find all components/services that are unmodified but whose module has been updated and add them to the modified array.
```

Comment:
`conf.Modified` was not exactly the right place to put these resources. Modular resources whose module has been reconfigured should really be treated as "newly added" (`conf.Added`) resources, so that the new, post-reconfiguration module process does not receive a `ReconfigureResourceRequest` and instead receives an `AddResourceRequest`.